src/main/java/de/propra/timetracker/Calculations.java

```diff
@@ -34,4 +34,12 @@ static int sumMinutesOfDate(List<Event> events, String datum) {
                 .mapToInt(Event::minuten)
                 .sum();
     }
+
+    static boolean isValidDate(String date) {
+        String validDateRegex = "^\\d{4}-" //YYYY-
+                                +"((((0[13578])|(1[02]))-(([0-2][0-9])|(3[01])))|" //MM-DD 31 Tage Monate
+                                +"(((0[469])|(11))-(([0-2][0-9])|(30)))|" //MM-DD 30 Tage Monate
+                                +"(02-[0-2][0-9]))$"; //MM-DD Februar (29 Tage Monat)
+        return date.matches(validDateRegex);
+    }
```

Comment:
macht es nicht eher auch sinn hier mit einem parser zu arbeiten? siehe Pull Request #24

Reply:
Mein erster Ansatz wäre gewesen, einfach versuchen das Datum zu parsen und die Exception zu fangen.
Man kann dann auch ruhig verschiedene Formate zulassen, der Parser kann ja einfach auf mehrere testen und wenn einer funktioniert, alles gut, wenn keiner funktioniert, dann sagt man dem User halt "ey korrigier deine Eingabe".

Reply:
Das lässt sich auch gut direkt in eine neue klasse auslagern

Reply:
Ein Regex ist hier nicht der richtige Einsatzzweck denke ich. Auch hier kann der selbe parser verwendet werden, wie bereits vorgeschlagen. Entspricht die Eingabe nicht dem Pattern im Parser, könnte eine passende Fehlermeldung geworfen und das Programm beendet werden.